main.go

```diff
@@ -0,0 +1,605 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/onflow/flow-go-sdk/access/grpc"
+)
+
+// TransactionArgs represents the arguments to construct a new transaction
+// or a message call.
+type TransactionArgs struct {
+	From                 *common.Address `json:"from"`
+	To                   *common.Address `json:"to"`
+	Gas                  *hexutil.Uint64 `json:"gas"`
+	GasPrice             *hexutil.Big    `json:"gasPrice"`
+	MaxFeePerGas         *hexutil.Big    `json:"maxFeePerGas"`
+	MaxPriorityFeePerGas *hexutil.Big    `json:"maxPriorityFeePerGas"`
+	Value                *hexutil.Big    `json:"value"`
+	Nonce                *hexutil.Uint64 `json:"nonce"`
+
+	// We accept "data" and "input" for backwards-compatibility reasons.
+	// "input" is the newer name and should be preferred by clients.
+	// Issue detail: https://github.com/ethereum/go-ethereum/issues/15628
+	Data  *hexutil.Bytes `json:"data"`
+	Input *hexutil.Bytes `json:"input"`
+
+	// Introduced by AccessListTxType transaction.
+	AccessList *types.AccessList `json:"accessList,omitempty"`
+	ChainID    *hexutil.Big      `json:"chainId,omitempty"`
+}
+
+type BlockChainAPI struct {
```

Comment:
I would say calling this grpcServer would be more appropriate, because in the future there might be multiple implementation of the "ethereum API".

Reply:
It would also be useful to define an interface for this API so you can mock out for testing (maybe too early for this tho)

Reply:
> I would say calling this grpcServer would be more appropriate, because in the future there might be multiple implementation of the "ethereum API".

This is not the actual JSON-RPC server, but an object that has to adhere to the Ethereum Specification. E.g.
```go
server := rpc.NewServer()
server.RegisterName("eth", &BlockChainAPI{})
```

Each exported method of this object (for example `BlockNumber`) is wired to an RPC method -> `eth_blockNumber`

> It would also be useful to define an interface for this API so you can mock out for testing (maybe too early for this tho)

That's indeed a good idea :+1: And I will also break it down to more types (`TransactionsAPI`, `AccountsAPI`, `FiltersAPI`).